### PR TITLE
[Addresing][Checkout] Choosing address from address book during checkout

### PR DESCRIPTION
**Pull request by Arminek**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Related tickets | -
| License         | MIT

### CODE REVIEW

File: features/checkout/addressing_order/choosing_billing_and_shipping_address_from_address_book.feature [hunk 1]
```diff
@@ -0,0 +1,34 @@
+@checkout
+Feature: Choosing an address from address book
+    In order to address an order by choosing it from my address book
+    As a customer
```
Comment (michalmarcinkowski) [inline features/checkout/addressing_order/choosing_billing_and_shipping_address_from_address_book.feature]: **C**ustomer

File: features/checkout/addressing_order/choosing_billing_and_shipping_address_from_address_book.feature [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    I want to be able to quick fill in my address information
```
Comment (michalmarcinkowski) [inline features/checkout/addressing_order/choosing_billing_and_shipping_address_from_address_book.feature]: This should be swapped with `In order to` 😄 

```
In order to quickly fill in my address information when addressing my order
...
I want to choose an address from my address book during checkout
```

File: features/checkout/addressing_order/choosing_billing_and_shipping_address_from_address_book.feature [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        And the store has a product "PHP T-Shirt" priced at "$19.99"
+        And I am a logged in customer
+        And I have an address "Lucifer Morningstar", "Seaside Fwy", "90802", "Los Angeles", "United States", "Arkansas" in my address book
+
+    @ui @javascript
+    Scenario: Choosing shipping address from address book
+        Given I have product "PHP T-Shirt" in the cart
+        And I am at the checkout addressing step
+        When I choose "Seaside Fwy" street for shipping address
+        Then I should have this address filled as shipping address
```
Comment (michalmarcinkowski) [inline features/checkout/addressing_order/choosing_billing_and_shipping_address_from_address_book.feature]: We shouldn't use `this address` and assert that the whole address is correct.

Comment (michalmarcinkowski) [inline features/checkout/addressing_order/choosing_billing_and_shipping_address_from_address_book.feature]: Then address "Lucifer Morningstar", "Seaside Fwy", "90802", "Los Angeles", "United States", "Arkansas" should be filled as shipping address

File: features/checkout/addressing_order/choosing_billing_and_shipping_address_from_address_book.feature [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    @ui @javascript
+    Scenario: Choosing billing address from address book
+        Given I have product "PHP T-Shirt" in the cart
+        And I am at the checkout addressing step
+        When I choose "Seaside Fwy" street for billing address
+        Then I should have this address filled as billing address
+
+    @ui @javascript
+    Scenario: Choosing shipping address from address book and proceed to the next step
+        Given I have product "PHP T-Shirt" in the cart
```
Comment (michalmarcinkowski) [inline features/checkout/addressing_order/choosing_billing_and_shipping_address_from_address_book.feature]: This step can be moved to the background.

File: src/Sylius/Behat/Context/Setup/CustomerContext.php [hunk 5]
```diff
@@ -182,6 +182,7 @@ public function iHaveAnAddressInAddressBook(ShopUserInterface $user, AddressInte
     {
         /** @var CustomerInterface $customer */
         $customer = $user->getCustomer();
+        $this->sharedStorage->set('address', $address);
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Setup/CustomerContext.php]: Will be redundant after https://github.com/Sylius/Sylius/pull/6526/files#r85093765

File: src/Sylius/Behat/Context/Transform/AddressingContext.php [hunk 6]
```diff
@@ -92,6 +102,17 @@ public function createNewAddressWithName($name, $street, $postcode, $city, $coun
     }
 
     /**
+     * @Transform /^"([^"]+)" street$/
+     */
+    public function getByStreet($street)
+    {
+        $address = $this->addressRepository->findOneBy(['street' => $street]);
+        Assert::notNull($address);
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Transform/AddressingContext.php]: Exception message should be specified.

File: src/Sylius/Behat/Page/Shop/Checkout/AddressPage.php [hunk 7]
```diff
@@ -322,4 +453,16 @@ private function waitForElement($timeout, $elementName)
...
+    /**
+     * @param string $type
+     */
+    private function assertAddressType($type)
+    {
+        $availableTypes = [self::TYPE_BILLING, self::TYPE_SHIPPING];
+
+        if (!in_array($type, $availableTypes, true)) {
+            throw new \InvalidArgumentException(sprintf('There are only two available types %s, %s. %s given', self::TYPE_BILLING, self::TYPE_SHIPPING, $type));
+        }
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Page/Shop/Checkout/AddressPage.php]: `Assert::oneOf`

**Comment by michalmarcinkowski** [comment]
Nice work Arek!

**Comment by pjedrzejewski** [comment]
👍 👍 👍

**Pull request merged**